### PR TITLE
Emails: Invite users to upgrade to Titan or Google Workspace on Email Plan page for email forwards

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request adds a new option at the bottom of the `Email Plan` page for email forwards in order to allow users to upgrade to a more powerful email solution such as Google Workspace or Titan:

This new link redirects to the `Email Comparison` page with a slightly different copy:

screenshot

Note this pull request also adds a missing page title to the `Email Plan` page, and makes sure that page titles from all email pages are correctly capitalized.

#### Testing instructions

1. Run `git checkout update/email-forwards-email-plan-page` and start your server, or open a [live branch](https://calypso.live/?branch=update/email-forwards-email-plan-page)
2. Log into a WordPress.com account that has a domain with email forwards
3. Open the [`Emails` page](http://calypso.localhost:3000/emails)
4. Click that domain to access the `Email forwarding settings` page
5. Click on the `Upgrade to a hosted email` navigation link
6. Assert that you are presented with the `Email Comparison` page
7. Assert that the title of the page is `Upgrade To A Hosted Email`
8. Assert that the title in the header is `Upgrade to a hosted email`
9. Assert that the `Email Forwarding` section is not displayed
10. Assert that you can purchase either a Titan or a Google Workspace account
11. Log into a WordPress.com account that has a domain with no emails this time
12. Navigate back to the `Emails` page
13. Click the `Add Email` button to access the `Email Comparison` page
14. Assert that the title of the page is `Add Email`
15. Assert that the title in the header is `Add email`
16. Assert that the `Email Forwarding` section is displayed

### CODE REVIEW

File: client/my-sites/email/email-providers-comparison/index.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-providers-comparison/index.jsx:96]: Since we need a `selectedSite` object for the header now, I've removed the `selectedSiteId` and `selectedSiteSlug` props since this data can be inferred from `selectedSite`.

**Comment by saygunnyc** [comment]
Thank you @stephanethomas for working on this!

I’ve done a bit of research to see about the right terminology, and it seems like we should call Google Workspace/Suite + Professional Email as hosted email solutions.

So, I am for using “**Upgrade to a hosted email**” over the CTA we have on the Email Forwarding detail page, as shared above.
For the upgrade page, I would repeat the CTA “**Upgrade to a hosted email**” on the Wayfinder so it connects well with the previous page.

For the banner area, I am for highlighting the difference between email forwarding and hosted email, as I believe the most value comes from **being able to send emails from that domain** aside from various features each product has.

Below is a quick mockup. Let me know your thoughts! cc @semihakocer

D - Email Comparison - WP comEmail-MultipleDomains

**Comment by stephanethomas** [comment]
Thank you for your feedback Saygun! I've updated the link and copy per your suggestion.

File: client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx
Comment (stephanethomas) [inline client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx:107]: This string has already been translated so there is limited benefit in using `titleCase()` here but at least we'll be consistent with other pages.

**Comment by stephanethomas** [comment]
By the way @saygunnyc , we used to display a notice when a user would specify the list of mailboxes to create whereas they already have email forwards set up (I believe that's something we forgot to implement in the new `Email Comparison` page):

screenshot

What do you think of adding a similar statement in the header?

> Pick from one of our flexible options to unlock full email features. Please note that email forwards are not compatible with those email solutions, and will be disabled once they are added to this domain.

Another option could be to show it under that header in a notice similar to the one above.

**Comment by semihakocer** [comment]
The proposed copy by @saygunnyc  above looks good to me. We've used hosted email wording on our logged-out pages as well.
@stephanethomas my 2 cents for the warning is to re-add it. I'd defer to you and Saygun on where to add it. It's very contextual and helpful one because their email forwards won't work anymore. One question: Do the customers need to setup anything after creating the email such that removing the email forward? If yes, we can also add a contextual message after users select an email and make sure they take those next steps.

**Comment by saygunnyc** [comment]
Good points @stephanethomas and @semihakocer  ! 

Someone who has multiple email forwards might think/expect upgrading will automatically convert their forwards into hosted mailboxes, however, that's not the world we live in today. 

I strongly believe there's an opportunity to improve this in the next iterations but we would need a different flow/design(s) to accomplish that. And, it is most likely better to track intent/traction with this first iteration before we go down that road, as I feel like it will be time-consuming.

@stephanethomas, I like your idea of adding it into the body copy as it doesn't create noise, but I'm concerned users might miss it there. So, I think **displaying a notice and making it clear that existing forwards will be removed** with the upgrade is I believe the way to go today. 

However, I am finding the copy on the existing notice to be lengthy and a bit hard to follow through.
How about if we say **"You can expect your existing email forwards to be removed once you switch to a hosted email"** on a notice card right under the banner in the interim?

Please let me know your thoughts and if another copy would work better?

**Comment by semihakocer** [comment]
@saygunnyc  @stephanethomas We probably need to mention which email forwards will be disabled in case users have multiple domains and email forwards. I'd tweak the copy to something like below to be more clear about the actions:

Your existing email forwards with the domain example.com will be removed once you switch to a hosted email solution. After the switch, make sure to setup your new email and don't miss a beat.

**Comment by daledupreez** [comment]
@semihakocer and @saygunnyc, I was working on those notices in #53463, and independently realised that we only want the notice to show on the email comparison page.

Can we comment on the notice page and wording on that PR?

As it stands, I've just updated the text in that other PR to:

> Please note that your existing email forwards for example.com will be removed if you upgrade to a hosted email solution. As part of the upgrade, you will need to specify the email addresses you want to keep using.

I am happy to tweak that as needed!

**Comment by stephanethomas** [comment]
I've rebased the branch with changes from https://github.com/Automattic/wp-calypso/pull/53463.

**Comment by stephanethomas** [comment]
Here is how the `Email Comparison` page looks like now:

screenshot

**Comment by saygunnyc** [comment]
> Please note that your existing email forwards for example.com will be removed if you upgrade to a hosted email solution. As part of the upgrade, you will need to specify the email addresses you want to keep using.

> Please note that your existing email forwards for example.com will be removed if you upgrade to a hosted email solution. Use the form below to set up the email addresses you want to continue using.

I like the **clarity of these** in directing to the form below for setup. 
I only wonder if we can try to cut this into a **single line** so it is easier to read, and visually looks better.

What if we use this, below?
"**Existing email forwards will be removed once you upgrade. Use the form below to set up the email addresses you want to continue using.**"
And then add the domain to the Wayfinder > **Upgrade to a hosted email: example.com**

Let me know your thoughts!

**Comment by saygunnyc** [comment]
One another thing I just noticed, @stephanethomas 
I think that would make sense to rename "Add Professional Email" to be "Upgrade to Professional Email"  to match the action user is actually taking. Let me know if that feels too long, then maybe we can just use "Upgrade"

The same goes for Google Workspace.

What are your thoughts?

**Comment by stephanethomas** [comment]
> I only wonder if we can try to cut this into a single line so it is easier to read, and visually looks better.

Well, don't forget that the length of this warning will vary depending on the language of the user anyway 🙂. Anyway, I've updated it per your suggestion, I've just kept the mention to the domain name as some users may have multiple domains with email forwards (so that it's clear which email forwards are going to be removed).

> And then add the domain to the Wayfinder > Upgrade to a hosted email: example.com

I have mixed feelings about this: the domain name is pretty clear in the promo card located just below, and we usually don't show the domain name in those `HeaderCake` components (we do it on the `Add New Mailboxes` pages but that's because we don't show a header like on the `Email Plan` page, and this is something we should most probably change).

> I think that would make sense to rename "Add Professional Email" to be "Upgrade to Professional Email" to match the action user is actually taking.

That's sounds like a good idea, I've updated all the buttons:

screenshot

**Comment by saygunnyc** [comment]
> That sounds like a good idea, I've updated all the buttons

Awesome! 🚀 

> Well, don't forget that the length of this warning will vary depending on the language of the user anyway.

I was with the assumption that the majority would have English as the preferred language, but perhaps I'm wrong. Is there an easy way to learn this? Not for this in particular but as useful knowledge for other things.

> I have mixed feelings about this: the domain name is pretty clear in the promo card located just below.

If that's the case then why do you think we would need it over the notification card? 😄

**Comment by stephanethomas** [comment]
> I was with the assumption that the majority would have English as the preferred language, but perhaps I'm wrong. Is there an easy way to learn this? 

I honestly don't know the answer but I'll try to find those numbers. I don't think we should design the user experience for English users only though, even if they represent a majority of our users (that would be a bit like saying that we don't care about accessibility).

> If that's the case then why do you think we would need it over the notification card? 😄

Well, for the reason I've mentioned earlier ☺: to make it crystal clear to users that we're only going to remove email forwards from that particular domain; otherwise they may wonder if other email forwards are going to be affected. But maybe I'm too careful, and it's true that most users have one domain on their site anyway.

**Comment by saygunnyc** [comment]
> I honestly don't know the answer but I'll try to find those numbers. I don't think we should design the user experience for English users only though, even if they represent a majority of our users (that would be a bit like saying that we don't care about accessibility).

Thank you so much for your help in digging through this @stephanethomas !! I am never for designing anything for only English users, but if we can show something better to the majority, I am for it. 

> Well, for the reason I've mentioned earlier ☺: to make it crystal clear to users that we're only going to remove email forwards from that particular domain; otherwise they may wonder if other email forwards are going to be affected. But maybe I'm too careful, and it's true that most users have one domain on their site anyway.

Now that we know **65% of users have English as their preferred language**, _–with your help_ 🙇‍♂️ and **with the knowledge around the majority of users having 1 domain**. Should we not take this as an advantage and have them see something better?

I'm honestly unsure about redundantly repeating something _– because then it starts to create clutter, rather than being helpful_ if we already communicate that over the header + the page that the user is coming from, also thinking about the overall context of the user thinking about an upgrade. I'd say they know what they are upgrading before they get into the path, but I can be wrong too.

No pressure here, wondering your thoughts. 
Let's do it if you agree, not too much of a problem if we don't. 😄

**Comment by stephanethomas** [comment]
> Should we not take this as an advantage and have them see something better?

Yes, that makes sense 👍.

File: client/my-sites/email/email-management/home/email-plan.jsx
Comment (olaseni) [inline client/my-sites/email/email-management/home/email-plan.jsx:379]: 👍

File: client/my-sites/email/email-providers-comparison/index.jsx
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx:155]: Would a one-liner suffice here?

```suggestion
	isUpgrading = () => hasEmailForwards( this.props.domain );
```

File: client/my-sites/email/email-providers-comparison/index.jsx
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: This can probably be the first statement as `isUpgrading()` doesn't actually take any arguments.
```suggestion
		if ( this.isUpgrading() ) {
```

File: client/my-sites/email/email-providers-comparison/index.jsx
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: Same sentiments about the domain argument being unneeded.

File: client/my-sites/email/email-existing-forwards-notice/index.jsx
Comment (olaseni) [inline client/my-sites/email/email-existing-forwards-notice/index.jsx]: _Non blocking nit_: Seeing that there could be "technically" multiple forms below (i.e. Google Workspace or Professional Email), do we want to be unequivocal in how we refer to the "form"? 

Am I right in thinking that the user could assume that the currently open form is what we ONLY refer to?

What about:
> Existing email forwards will be removed once you upgrade. Set up the email addresses you want to continue using below ?

I could be spiralling but you can weigh in here @saygunnyc, @semihakocer

**Review by olaseni — APPROVED** [review]
This looks mostly good to go even though I had some non-blocking suggestions. 

There are some needed changes before merging, but I don't think another round of review is needed.

**Comment by saygunnyc** [comment]
Thank you, @olaseni !

> Am I right in thinking that the user could assume that the currently open form is what we ONLY refer to?

Good point. Not fully sure if this would lead to that as we also say "pick from our flexible options" above.  

> Existing email forwards will be removed once you upgrade. Set up the email addresses you want to continue using below.

I might, so this also makes sense to me. What do you think @semihakocer ?

File: client/my-sites/email/email-providers-comparison/index.jsx
Comment (stephanethomas) [reply client/my-sites/email/email-providers-comparison/index.jsx]: Good catch!

File: client/my-sites/email/email-providers-comparison/index.jsx
Comment (stephanethomas) [reply client/my-sites/email/email-providers-comparison/index.jsx]: I found some other places with the same issue 🤦‍♂️.

**Comment by stephanethomas** [comment]
I've rebased to fix a merge conflicts, and addressed all comments.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6096637

Thank you @stephanethomas for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6096637).

**Pull request merged**
